oak-core/src/test/java/org/apache/jackrabbit/oak/query/AbstractQueryTest.java

```diff
@@ -344,13 +345,21 @@ protected static void assertResultSorted(@NotNull List<String> expected, @NotNul
         assertEquals("Expected sorted result not found", expected, actual);
     }
 
+    /**
+     * Checks that the two lists have the same elements ignoring their order.
+     */
     protected static void assertResult(@NotNull List<String> expected, @NotNull List<String> actual) {
-        for (String p : checkNotNull(expected)) {
-            assertTrue("Expected path " + p + " not found, got " + actual, checkNotNull(actual)
-                    .contains(p));
-        }
-        assertEquals("Result set size is different: " + actual, expected.size(),
-                actual.size());
+        checkNotNull(expected);
+        checkNotNull(actual);
+        assertEquals("Result set size is different. Expected: " + expected + ", actual: " + actual, expected.size(), actual.size());
+
+        String[] sortedActual = actual.toArray(new String[0]);
+        Arrays.sort(sortedActual);
+
+        String[] sortedExpected = expected.toArray(new String[0]);
+        Arrays.sort(sortedExpected);
+
+        assertTrue("Invalid result set. Expected: " + expected + ", Actual: " + actual, Arrays.equals(sortedExpected, sortedActual));
```

Comment:
this can be simplified
```suggestion
        assertArrayEquals("Invalid result set. Expected: " + expected + ", Actual: " + actual, sortedExpected, sortedActual);
```